File: frontend/src/types/telemetry.ts
export interface GPSCoord {
  lat: number;
  lng: number;
}

export interface DroneTelemetry {
  timestamp: string;
  gps: GPSCoord;
  altitude_m: number;
  speed_mps: number;
  heading_deg: number;
  battery_pct: number;
  signal_strength: number;
  camera_mode: string;
}

File: backend/schemas/telemetry.py
from datetime import datetime
from pydantic import BaseModel


class GPSCoord(BaseModel):
    lat: float
    lng: float


class DroneTelemetry(BaseModel):
    timestamp: datetime
    gps: GPSCoord
    altitude_m: float
    speed_mps: float
    heading_deg: float
    battery_pct: int
    signal_strength: int
    camera_mode: str  # "thermal" | "rgb"
